cfme/tests/cloud_infra_common/test_provisioning.py

```diff
@@ -204,32 +229,17 @@ def test_provision_approval(appliance, provider, vm_name, smtp_test, request,
     }
 
     vm = collection.create(vm_name, provider, form_values=inst_args, wait=False)
-    try:
-        wait_for(
-            lambda: len(smtp_test.get_emails()) >= 2,
-            num_sec=90,
-            delay=3
-        )
-    except TimedOutError:
-        pytest.fail('Did not receive at least 2 emails from provisioning request, received: {}'
-                    .format(smtp_test.get_emails()))
-
     pending_subject = normalize_text("your {} request is pending".format(vm_type))
     # requester includes the trailing space
     pending_from = normalize_text("{} request from {}pending approval".format(vm_type, requester))
 
-    received_pending = [normalize_text(m["subject"]) for m in smtp_test.get_emails()]
-    # Looking for each expected subject in the list of received subjects with partial match
-    for subject in [pending_subject, pending_from]:
-        soft_assert(any(subject in r_sub for r_sub in received_pending),
-                    'Expected subject [{}], not matched in received subjects [{}]'
-                    .format(subject, received_pending))
+    wait_for_messages_with_subjects(smtp_test, [pending_subject, pending_from], num_sec=90)
 
     smtp_test.clear_database()
 
     cells = {'Description': 'Provision from [{}] to [{}###]'.format(vm.template_name, vm.name)}
-    provision_request = appliance.collections.requests.instantiate(cells=cells)
-    if edit:
+
+    def action_edit():
```

Comment:
To keep the test structure more transparent, I think it'd be helpful if these `action_` function were defined out of the test case itself. What do you think?

Reply:
I am not sure what is better but I have not much against putting them out except that Surely I will need to pass more arguments to them.

I kept those `actions_*` as I didn't think they are reusable anywhere else.

Reply:
Okay, I'm fine with keeping them in the test function, I'd consider my comment _optional_

Reply:
Agreed @john-dupuy, but I think this is optional to this PR.